mslice/cli/helperfunctions.py

```diff
@@ -4,6 +4,68 @@
 from mslice.models.alg_workspace_ops import get_axis_range, get_available_axes
 from mslice.models.axis import Axis
 from mslice.models.workspacemanager.workspace_provider import workspace_exists
+from mslice.plotting.globalfiguremanager import GlobalFigureManager
+
+_overplot_keys = {'Hydrogen': 1, 'Deuterium': 2, 'Helium': 4, 'Aluminium': 'Aluminium', 'Copper': 'Copper',
+                  'Niobium': 'Niobium', 'Tantalum': 'Tantalum', 'Arbitrary Nuclei': 'Arbitrary Nuclei',
+                  'CIF file': 'CIF file'}
+
+_function_to_intensity = {
+    'show_scattering_function': 's(q,e)',
+    'show_dynamical_susceptibility': 'chi',
+    'show_dynamical_susceptibility_magnetic': 'chi_mag',
+    'show_d2sigma': 'xsec',
+    'show_symmetrised': 'symm',
+    'show_gdos': 'gdos',
+}
+
+_intensity_to_action = {
+    's(q,e)': 'action_sqe',
+    'chi': 'action_chi_qe',
+    'chi_mag': 'action_chi_qe_magnetic',
+    'xsec': 'action_d2sig_dw_de',
+    'symm': 'action_symmetrised_sqe',
+    'gdos': 'action_gdos',
+}
+
+_intensity_to_workspace = {
+    'chi': 'chi',
+    'chi_mag': 'chi_magnetic',
+    'xsec': 'd2sigma',
+    'symm': 'symmetrised',
+    'gdos': 'gdos',
+}
+
+
+def _update_legend():
+    plot_handler = GlobalFigureManager.get_active_figure()._plot_handler
+    plot_handler.update_legend()
```

Comment:
I am unable to reproduce this error, @mducle 
Could I have access to the same file? Or perhaps you could try again when I push to github again?

Reply:
I'll try it again when you've pushed to github... I can't seem to upload the file to github but it's on my folder on: \\Olympic\Babylon5\Public\duc\

Reply:
Ok... This is very weird - so when I run the script externally, e.g. with: 

```
mantidpython test1.py
```

it works fine (PS - you can just use an input file you have rather than the one I used - it doesn't matter which file it is for this error).

However, when I try to run the file within Mantidplot or within the IPython console, using e.g.

```
import test1
```

it fails with the `plot_handler` = `None` problem. The same error occurs if I type all the commands into the IPython console, or if I run the file from the Python script window in MantidPlot (accessible using `F3`) in Mantidplot *after* I start the MSlice interface. If I run the file from the script window in MantidPlot *before* starting the MSlice interface it works fine...

So I think there is something in the MSlice GUI initialisation which is causing the problem...